cmd/kwild/server/migration.go

```diff
@@ -139,33 +130,40 @@ func (m *migrationClient) downloadGenesisState(ctx context.Context) error {
 		return fmt.Errorf("failed to unmarshal genesis config: %w", err)
 	}
 
-	// Save the genesis state
-	var snapshotMetadata statesync.Snapshot
-	if err := json.Unmarshal(metadata.SnapshotMetadata, &snapshotMetadata); err != nil {
-		return fmt.Errorf("failed to unmarshal snapshot metadata: %w", err)
-	}
+	genesisStateExists := bytes.Equal(genCfg.DataAppHash, m.genesisCfg.DataAppHash) &&
+		validateGenesisState(m.snapshotFileName, genCfg.DataAppHash)
```

Comment:
In 

```go
func validateGenesisState(filename string, appHash []byte) bool {
	// check if the genesis state file exists
	if _, err := os.Stat(filename); os.IsNotExist(err) {
		return false
	}

	genesisStateFile, err := os.Open(filename)
	if err != nil {
		return false
	}
```

Shouldn't that error with `os.Open` cause `failBuild`?

Reply:
Why, if we can regenerate the file ?

Reply:
Maybe I just misunderstand the intent of `validateGenesisState`.  If the file the file does not exist, OK.  If it exists, but `os.Open` fails, OK?  If open succeeds but gunzip or io.Copy errors (which would be the if the file is not readable too), NOT OK? I don't see why this last case should be OK but it's ok to ignore an error form `os.Open`.  Either case indicates something is wrong with the file.

Reply:
> Why, if we can regenerate the file with correct permissions?

There's only a narrow chance of that working.  If it's not owned by you, you can't just recreated it with correct permissions.  Errors other than permissions will also prevent os.Create from working.

Fundamentally, if a the file exists, but it cannot be read, that's clearly not what is expected and should be treated as a corrupt file just as if gunzip or io.Copy were to fail.